packages/dialog/index.tsx

```diff
@@ -0,0 +1,322 @@
+    const {closest, matches} = ponyfill;
+    const {isScrollable, areTopsMisaligned} = util;
+    return {
+      addClass: (className: string) => {
+        const {classList} = this.state;
+        classList.add(className);
+        this.setState({classList});
+      },
+      removeClass: (className: string) => {
+        const {classList} = this.state;
+        classList.delete(className);
+        this.setState({classList});
+      },
+      hasClass: (className: string) => this.classes.split(' ').includes(className),
+      addBodyClass: (className: string) => document.body.classList.add(className),
+      removeBodyClass: (className: string) => document.body.classList.remove(className),
+      eventTargetMatches: (target: HTMLElement, selector: string) => matches(target, selector),
+      trapFocus: () => this.focusTrap && this.focusTrap.activate(),
+      releaseFocus: () => this.focusTrap && this.focusTrap.deactivate(),
+      isContentScrollable: () => {
+        const content = this.content_;
+        return (!!content) ? isScrollable(content) : false;
+      },
+      areButtonsStacked: () => areTopsMisaligned(this.buttons_),
+      getActionFromEvent: (evt: any) => {
+        const elem = closest(evt.target, `[${strings.ACTION_ATTRIBUTE}]`);
+        return elem && elem.getAttribute(strings.ACTION_ATTRIBUTE);
+      },
+      clickDefaultButton: () => {
+        const defaultButton = this.defaultButton_;
+        if (defaultButton) {
+          defaultButton.click();
+        }
+      },
+      reverseButtons: () => {
+        const buttons = this.buttons_;
+        return buttons &&
+          buttons.reverse().forEach((button: HTMLButtonElement) =>
+            button.parentElement && button.parentElement.appendChild(button)
+          );
+      },
+      notifyOpening: () => this.handleOpening(),
+      notifyOpened: () => this.props.onOpen && this.props.onOpen(),
+      notifyClosing: (action: string) => this.handleClosing(action),
+      notifyClosed: (action: string) => this.props.onClose && this.props.onClose(action),
+    };
+  }
+
+  handleOpening = (): void => {
+    this.props.onOpening!();
+    LAYOUT_EVENTS.forEach((evt: string) =>
+      window.addEventListener(evt, this.handleLayout)
+    );
+    document.addEventListener(
+      'keydown', this.handleDocumentKeyDown
+    );
+  }
+
+  handleClosing = (action: string): void => {
+    this.props.onClosing!(action);
+    LAYOUT_EVENTS.forEach((evt: string) =>
+      window.removeEventListener(evt, this.handleLayout)
+    );
+    document.removeEventListener(
+      'keydown', this.handleDocumentKeyDown
+    );
+  }
+
+  handleInteraction = (e: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>): void =>
+    this.foundation.handleInteraction(e);
+  handleDocumentKeyDown = (e: Event): void =>
+    this.foundation.handleDocumentKeydown(e);
+  handleLayout = (): void => this.foundation.layout();
+
+
+  render() {
+    const {
```

Comment:
add onClick and onKeydown here, so otherProps does not override dialog's methods.

`this.handleInteraction` should call `this.props.onClick` or `this.props.onKeydown` if the event was a click or key event.